src/printer/index.js

```diff
@@ -608,12 +597,26 @@ function genericPrint(path, options, print) {
     case "arg_paren": {
       return path.call(print, "args");
     }
+
+    case "regexp_literal":
+      return concat([
+        "/",
+        join("", path.map(print, "content")),
+        "/",
+        path.call(print, "regex_end")
+      ]);
+
+    case "regexp_end":
+      return path.call(print, "regexp_end");
+
     case "string_literal": {
-      return concat(path.map(print, "string_content"));
+      const rawContent = n.string_content[0].content;
+
+      return printRubyString(rawContent, options);
     }
 
     case "@tstring_content": {
-      return printRubyString('"' + path.call(print, "content") + '"', options);
+      return path.call(print, "content");
```

Comment:
`@tstring_content` I removed quoting at this level, as a Ruby regexp has `@tstring_content` but we wouldn't want to quote it:

```
    console.log src/parser.js:26
      tokens:
       [[[1, 0], :on_regexp_beg, "/"],
       [[1, 1], :on_tstring_content, "\\A[\\w+-_]+\\Z"],
       [[1, 13], :on_regexp_end, "/mi"],
       [[1, 16], :on_nl, "\n"]]

    console.log src/parser.js:27
      sexp:
       [:program,
       [[:regexp_literal,
         [[:@tstring_content, "\\A[\\w+-_]+\\Z", [1, 1]]],
         [:@regexp_end, "/mi", [1, 13]]]]]

    console.log src/parser.js:28
      json AST:
       {
          "ast_type": "program",
          "body": [
              {
                  "ast_type": "regexp_literal",
                  "content": [
                      {
                          "ast_type": "@tstring_content",
                          "content": "\\A[\\w+-_]+\\Z"
                      }
                  ],
                  "regexp_end": {
                      "ast_type": "@regexp_end",
                      "regexp_end": "/mi"
                  }
              }
          ],
          "comments": []
      }
```

Reply:
I also noticed that the string interpolation syntax and the`%w( ... )` notation also has similar issues when trying to format it if `@tstring_content` assumes it is quoted